lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -41,6 +41,38 @@ def object
           end
         end
 
+        # Parse a Hash or ActionController::Parameters representing a JSON API document
+        # into an ActiveRecord-ready hash.
+        # NOTE(beauby): Currently this does not handle relationships with modified keys.
+        #
+        # @param [Hash|ActionController::Parameters] document
+        # @return [Hash] ActiveRecord-ready hash
+        #
+        def self.parse(document)
+          hash = {}
+
+          hash[:id] = document['data']['id'] if document['data']['id']
```

Comment:
Can we try to stick to the ActiveModel::Serializers::Json API
https://github.com/rails/rails/blob/4-2-stable/activemodel/lib/active_model/serializers/json.rb#L137
`from_json`?

For completeness I'd also like to compare this to other libraries out there
and what language they use.

I think, as we've discussed, there should be a DeserializableDocument class
that has a from_json method on it, and that perhaps uses a Deserializer
underneath ref:
https://github.com/rails-api/active_model_serializers/issues/1098